### PR TITLE
Adds a pricing API for NFTs

### PR DESCRIPTION
**Pull request by nicarq**
Adds a pricing API for NFTs e.g.

```
## CNFT Prices
curl -X "POST" "http://localhost:8080/v1/getNFTPrices" \
     -H 'Accept-Encoding: deflate, gzip;q=1.0, *;q=0.5' \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'[
  "b97859c71e4e73af3ae83c30a3172c434c43041f6ff19c297fb76094",
  "b863bc7369f46136ac1048adb2fa7dae3af944c3bbb2be2f216a8d4f",
  "79bf686fb621898a78fe8ee56825028a2cd50a79d408d290f89faf22",
  "40fa2aa67258b4ce7b5782f74831d46a84c59a0ff0c28262fab21728"
]'
```

Returns:

```
{
  "status": "ok",
  "data": [
    {
      "id": 63760,
      "name": "EarthNode",
      "policies": "b97859c71e4e73af3ae83c30a3172c434c43041f6ff19c297fb76094",
      "data": {
        "attribution": "NOTICE: © 2022 OpenCNFT. Just include the source of the data (opencnft.io) ;)",
        "policy": "b97859c71e4e73af3ae83c30a3172c434c43041f6ff19c297fb76094",
        "thumbnail": "ipfs://QmYUA3THPRhQcdmSzgJUTYMa98hvYWtD9i9vUG8ZcNeQeo",
        "total_volume": 2745353000000,
        "first_sale": 1.7976931348623157e+308,
        "total_tx": 33,
        "total_assets_sold": 32,
        "asset_minted": 1000,
        "asset_holders": 515,
        "highest_sale": {
          "asset_name": "EarthNode835",
          "name": "b97859c71e4e73af3ae83c30a3172c434c43041f6ff19c297fb76094.EarthNode835",
          "price": 135000000000
        },
        "floor_price": 78500000000,
        "floor_price_marketplace": "jpg.store"
      },
      "lastUpdatedTimestamp": 1675826236512
    },
    {
      "id": 27835,
      "name": "Berry",
      "policies": "b863bc7369f46136ac1048adb2fa7dae3af944c3bbb2be2f216a8d4f",
      "data": {
        "attribution": "NOTICE: © 2022 OpenCNFT. Just include the source of the data (opencnft.io) ;)",
        "policy": "b863bc7369f46136ac1048adb2fa7dae3af944c3bbb2be2f216a8d4f",
        "thumbnail": "ipfs://ipfs/Qmb43sxTNDVze7VAodazKoPDaB6vjMKs8sEqGuexiAu5TS",
        "total_volume": 1186259990000,
        "first_sale": 1.7976931348623157e+308,
        "total_tx": 16,
        "total_assets_sold": 16,
        "asset_minted": 100,
        "asset_holders": 84,
        "highest_sale": {
          "asset_name": "BerryCardinal",
          "name": "b863bc7369f46136ac1048adb2fa7dae3af944c3bbb2be2f216a8d4f.BerryCardinal",
          "price": 115000000000
        },
        "floor_price": 64000000000,
        "floor_price_marketplace": "jpg.store"
      },
      "lastUpdatedTimestamp": 1675826236850
    },
    {
      "id": 64080,
      "name": "Normandy",
      "policies": "79bf686fb621898a78fe8ee56825028a2cd50a79d408d290f89faf22",
      "data": {
        "attribution": "NOTICE: © 2022 OpenCNFT. Just include the source of the data (opencnft.io) ;)",
        "policy": "79bf686fb621898a78fe8ee56825028a2cd50a79d408d290f89faf22",
        "thumbnail": "ipfs://QmXKgqBWtNRuYqGVuRJaxGJsTZxLBb9ypBPDqjBMhVHJAB",
        "total_volume": 90055000000,
        "first_sale": 1.7976931348623157e+308,
        "total_tx": 4,
        "total_assets_sold": 4,
        "asset_minted": 8,
        "asset_holders": 2,
        "highest_sale": {
          "asset_name": "AMemoryofLight",
          "name": "79bf686fb621898a78fe8ee56825028a2cd50a79d408d290f89faf22.AMemoryofLight",
          "price": 55555000000
        },
        "floor_price": 11500000000,
        "floor_price_marketplace": "jpg.store"
      },
      "lastUpdatedTimestamp": 1675826237504
    },
    {
      "id": 7867,
      "name": "Clay Nation",
      "policies": "40fa2aa67258b4ce7b5782f74831d46a84c59a0ff0c28262fab21728",
      "data": {
        "attribution": "NOTICE: © 2022 OpenCNFT. Just include the source of the data (opencnft.io) ;)",
        "policy": "40fa2aa67258b4ce7b5782f74831d46a84c59a0ff0c28262fab21728",
        "thumbnail": "ipfs://QmfZWQPomygx3E7qE3R8k6LVGGPhCN1gnCkk2TxrZk6pQ1",
        "total_volume": 32876131362782,
        "first_sale": 1.7976931348623157e+308,
        "total_tx": 17400,
        "total_assets_sold": 7284,
        "asset_minted": 9970,
        "asset_holders": 4284,
        "highest_sale": {
          "asset_name": "ClayNation1861",
          "name": "40fa2aa67258b4ce7b5782f74831d46a84c59a0ff0c28262fab21728.ClayNation1861",
          "price": 100000000000
        },
        "floor_price": 3983000000,
        "floor_price_marketplace": "jpg.store"
      },
      "lastUpdatedTimestamp": 1675826237771
    }
  ]
}
```

### CODE REVIEW

File: tests/nftPrice.test.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+          currentCNFTsPrice[
+            "b863bc7369f46136ac1048adb2fa7dae3af944c3bbb2be2f216a8d4f"
+          ].lastUpdatedTimestamp,
+      });
+    } catch (e) {
+      throw e;
+    }
+  });
+
+  // TODO: fix the nock error for this test
```
Comment (SebastienGllmt) [inline tests/nftPrice.test.ts:148]: TODO?

File: src/opencnft.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const fetcher = logging ? axios.get : axiosRaw.get;
+  return fetcher(getCollectionsURL()).then((resp) => {
+    if (resp.status === 404) {
+      throw new Error("Not found");
+    }
+    return parseCNFTPredatorResponse(resp.data);
+  });
+};
+
+export const parseCNFTResponse = (resp: any): CNFT => {
```
Comment (SebastienGllmt) [inline src/opencnft.ts:36]: maybe `unknown` instead of `any` here

File: src/index.ts [hunk 3]
```diff
@@ -226,74 +240,112 @@ const getPriceEndpoint = async (req: Request, res: Response) => {
...
+  { path: "/v1/getNFTPrices", method: "post", handler: getNFTsPriceEndpoint },
 ];
 
 applyRoutes(routes, router);
 router.use(middleware.logErrors);
 router.use(middleware.errorHandler);
 
+router.get("/", (req, res) => {
+  res.send("Hello World!");
+});
```
Comment (SebastienGllmt) [inline src/index.ts:278]: Is this still needed?

**Comment by SebastienGllmt** [comment]
We may be able to delete the deprecated token registry from the token-registry-service: https://github.com/dcSpark/token-registry-service/tree/main/registry

File: tests/nftPrice.test.ts (hunk 1 above)
Comment (nicarq) [reply tests/nftPrice.test.ts:148]: yeah for another PR. I couldn't figure it out and it's not required. No idea why Nock is acting weird

File: src/opencnft.ts (hunk 2 above)
Comment (nicarq) [reply src/opencnft.ts:36]: maybe @ggaabe can improve it

File: src/index.ts (hunk 3 above)
Comment (nicarq) [reply src/index.ts:278]: ah we can remove this

**Pull request merged**